src/cmd/main.go

```diff
@@ -2,30 +2,103 @@ package main
+	KubernetesStateChecker []checker.Check `yaml:"kubernetes-state-checker"`
+}
+
+func (c *conf) getConf() *conf {
+
+	yamlFile, err := ioutil.ReadFile("conf.yaml")
+	if err != nil {
+		log.Printf("yamlFile.Get err   #%v ", err)
+	}
+	err = yaml.Unmarshal(yamlFile, c)
+	if err != nil {
+		log.Fatalf("Unmarshal: %v", err)
+	}
+
+	return c
+}
+
 func main() {
 
 	// Get kubeconfig
+	// kubeconfig setup example: https://github.com/kubernetes/client-go/blob/master/examples/out-of-cluster-client-configuration/main.go
+	var kubeconfig *string
+	if home := homedir.HomeDir(); home != "" {
+		kubeconfig = flag.String("kubeconfig", filepath.Join(home, ".kube", "config"), "(optional) absolute path to the kubeconfig file")
+	} else {
+		kubeconfig = flag.String("kubeconfig", "", "absolute path to the kubeconfig file")
+	}
+	flag.Parse()
+
+	// use the current context in kubeconfig
+	config, err := clientcmd.BuildConfigFromFlags("", *kubeconfig)
+	if err != nil {
+		panic(err.Error())
+	}
+
+	// create the clientset
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		panic(err.Error())
+	}
 
 	// Get input yaml with checks
+	var c conf
+	c.getConf()
+
+	for _, aCheck := range c.KubernetesStateChecker {
+
+		// convert to yaml
+		valuesYaml, err := yaml.Marshal(aCheck)
+		if err != nil {
+			panic(err.Error())
+		}
 
-	// Execute the check runner
-	c := checker.New(
-		 "doesServicePortExist",
-		  "Does microservice 1 have a kubernetes service with port 5000 exposed",
-		  "This checks if microservice 1 has a Kubernetes service with port 5000 exposed",
-		  "app",
-	)
-	results := c.Run()
-
-	fmt.Println(fmt.Sprintf(`+----------------------------------------------------------------------+------+-------------+	
-| Name                                                                 | Pass | Message     |
-+----------------------------------------------------------------------+------+-------------+
-| %s | %s | %s  |
-+----------------------------------------------------------------------+------+-------------+`,
-		name, strconv.FormatBool(results.DidPass), results.Message))
+		// Execute the check runner
+		chk := checker.New(
+			BytesToString(valuesYaml),
+			clientset,
+			aCheck.Ttype,
+			aCheck.Name,
+			aCheck.Description,
+			aCheck.Namespace,
+			aCheck.Values,
+		)
+		results := chk.Run()
+
+		fmt.Println(fmt.Sprintf(`+----------------------+----------------------------------------------------------------------+------+-------------+
+	| Test Type          | Name                                                                 | Pass | Message     |
+	+----------------------+----------------------------------------------------------------------+------+-------------+
+	| %s | %s | %s | %s  |
+	+----------------------+----------------------------------------------------------------------+------+-------------+`,
+			aCheck.Ttype, c.KubernetesStateChecker[0].Name, strconv.FormatBool(results.DidPass), results.Message))
```

Comment:
Will make this dynamic after this PR.  Not exactly sure how we want the output yet.

Reply:
Something like k8s.io/client-go/util/homedir might help

Reply:
Thanks.  @mazamats any thoughts on what this output should contain?  I thought about it a little bit but not that much.

Reply:
I think we could mimic a jest style test output.

I'm thinking you could display color coded `PASS/FAIL` in the table and then expand on failed tests below the table.

Could possibly even diff the YAML spec with expected and actual like the image below. I'm not sure if this could be used internally https://github.com/homeport/dyff

Example: 
![image](https://user-images.githubusercontent.com/25627421/100148089-826bc500-2e59-11eb-8007-552af969ab98.png)

Reply:
I captured the info on the output suggestions in this ticket which is for designing the output: https://anthemai.atlassian.net/browse/HSS-184